### PR TITLE
fix/use a service's fixed phone number if set

### PR DESCRIPTION
**Pull request by sastels**
# Summary | Résumé

Have a case where a [service with a dedicated number is not using the number for bulk sends](https://cds-snc.freshdesk.com/a/tickets/17255) though it is for individual sends.

Found that the `reply_to_text` is not being set for these notifications. Admin one-offs still are working fine, but the other 3 ways to send are affected.

I think there's an issue where foreign keys set in other tables but referencing the `services` table do not appear in the services table when we cache it. In this case, though, we can just set the reply_to_text / sender_id when the job or notification is created.

# Test instructions | Instructions pour tester la modification

- Set a dedicated phone number for a service (locally you can just choose one from staging and add it to your local setup as an inbound number, then assign it to your service)
- send yourself some sms. All should come from the dedicated number.
- go to another service that doesn't have a set number
- send yourself sms
- all should come from a number different from the dedicated number (ie one of our numbers in ca-central-1)

# Release Instructions | Instructions pour le déploiement

Ask [user](https://cds-snc.freshdesk.com/a/tickets/17255) to test.

# Reviewer checklist | Liste de vérification du réviseur

- [ ] This PR does not break existing functionality.
- [ ] This PR does not violate GCNotify's privacy policies.
- [ ] This PR does not raise new security concerns. Refer to our GC Notify Risk Register document on our Google drive.
- [ ] This PR does not significantly alter performance.
- [ ] Additional required documentation resulting of these changes is covered (such as the README, setup instructions, a related ADR or the technical documentation).

> ⚠ If boxes cannot be checked off before merging the PR, they should be moved to the "Release Instructions" section with appropriate steps required to verify before release. For example, changes to celery code may require tests on staging to verify that performance has not been affected.

### CODE REVIEW

File: app/celery/tasks.py [hunk 1]
```diff
@@ -245,7 +248,9 @@ def save_smss(self, service_id: Optional[str], signed_notifications: List[Signed
 
         reply_to_text = ""  # type: ignore
         if sender_id:
-            reply_to_text = dao_get_service_sms_senders_by_id(service_id, sender_id).sms_sender
+            reply_to_text = try_validate_and_format_phone_number(
+                dao_get_service_sms_senders_by_id(service_id, sender_id).sms_sender
+            )
```
Comment (sastels) [inline app/celery/tasks.py:253]: The old code is wrong - later on we ignore the reply_to_text if it doesn't start with a "+".

File: app/celery/tasks.py [hunk 2]
```diff
@@ -243,13 +246,18 @@ def save_smss(self, service_id: Optional[str], signed_notifications: List[Signed
         sender_id = _notification.get("sender_id")  # type: ignore
         notification_id = _notification.get("id", create_uuid())
 
-        reply_to_text = ""  # type: ignore
-        if sender_id:
-            reply_to_text = dao_get_service_sms_senders_by_id(service_id, sender_id).sms_sender
-        elif template.service:
-            reply_to_text = template.get_reply_to_text()
+        if "reply_to_text" in _notification and _notification["reply_to_text"]:
+            reply_to_text = _notification["reply_to_text"]
```
Comment (sastels) [inline app/celery/tasks.py:250]: The code allows us to pass it in, but we were explicitly ignoring the passed in value (?)

**Review by whabanks — APPROVED** [review]
LGTM - Tested locally per instructions and both one offs and bulk sends are using the dedicated number as expected! 🚀

**Review by jimleroyer — COMMENTED** [review]
Can we add some tests with this new scenario where the sender IDs would be null and needs to be applied?

**Review by jimleroyer — APPROVED** [review]
LGTM!

**Comment by sastels** [comment]
Will merge and test tomorrow!
